iofog-agent-daemon/src/test/java/org/eclipse/iofog/command_line/CommandLineActionTest.java

```diff
@@ -0,0 +1,408 @@
+
+    }
+    /**
+     * When config command without options display help
+     */
+    @Test
+    public void configActionPerformWithInvalidOption() throws Exception {
+
+        // Test when config option value is ambiguous
+        String[] anotherArgs = {"config", "ambiguous"};
+        assertEquals(helpContent, CommandLineAction.getActionByKey(anotherArgs[0]).perform(anotherArgs));
+
+    }
+
+    /**
+     * When config command with default to reset the configuration Success
+     */
+    @Test
+    public void configActionPerformWithDefaultOption() throws Exception {
+
+        // Test when config reset to defaults
+        String[] anotherArgs1 = {"config", "defaults"};
+        assertEquals("Configuration has been reset to its defaults.",
+                CommandLineAction.getActionByKey(anotherArgs1[0]).perform(anotherArgs1));
+
+    }
+
+    /**
+     * When config command with valid option with no value
+     */
+    @Test
+    public void configActionPerformWithValidOptionNoValue() throws Exception {
+
+        // Test when config reset to defaults
+        String[] logArgs = {"config", "-ll"};
+        assertEquals(helpContent, CommandLineAction.getActionByKey(logArgs[0]).perform(logArgs));
+
+
+    }
+
+    /**
+     * When config command with valid option with InValid value then Configuration throw exception
+     */
+    @Test
+    public void configActionPerformWithValidOptionInvalidValue() throws Exception {
+
+        // Test when config reset to defaults
+        String[] logArgs = {"config", "-ll", "severeAndInfo"};
+
+        PowerMockito.when(Configuration.getOldNodeValuesForParameters(anySet(), any())).
+                thenReturn(result);
+        PowerMockito.when(Configuration.setConfig(anyMap(), anyBoolean())).thenThrow(new Exception("item not found or defined more than once"));
+
+        assertEquals("Error updating new config : item not found or defined more than once",
+                CommandLineAction.getActionByKey(logArgs[0]).perform(logArgs));
+
+    }
+
+
+    /**
+     * When config command with valid option and value
+     */
+    @Test
+    public void configActionPerformWithValidOptionAndValue() throws Exception {
+
+        String[] logArgs = {"config", "-ll", "severe"};
+        PowerMockito.when(Configuration.setConfig(anyMap(), anyBoolean())).thenReturn(new HashMap<>());
+        PowerMockito.when(Configuration.getOldNodeValuesForParameters(anySet(), any())).
+                thenReturn(result);
+
+        assertEquals("\\n\tChange accepted for Parameter : - ll, Old value was :info, New Value is : severe",
+                CommandLineAction.getActionByKey(logArgs[0]).perform(logArgs));
+
+    }
+
+    /**
+     * Helper method for comparing lists
+     */
+    private static boolean isEqual(List list1, List list2) {
+        boolean value = list1.size() == list2.size() && list1.equals(list2);
+        return value;
+    }
+
+    private String version = "ioFog 1.2.2 \n" +
```

Comment:
Looks like you need to mock `CmdProperties#getVersion()` method as well. What would happen when we bump the version? My guess is that the version test fails, right?

Reply:
Yeah true get version will fail. Will try mock getVersion and make the result independent.

Reply:
Updated